tests/gpflow/conditionals/test_multioutput.py

```diff
@@ -138,66 +162,75 @@ def create_q_sqrt(M: int, L: int) -> AnyNDArray:
 
 
 class Data:
+    cs = ShapeChecker().check_shape
+
     N, Ntest = 20, 5
     D = 1  # input dimension
     M = 3  # inducing points
     L = 2  # latent gps
     P = 3  # output dimension
     MAXITER = int(15e2)
-    X = tf.random.normal((N,), dtype=tf.float64)[:, None] * 10 - 5
-    G: AnyNDArray = np.hstack((0.5 * np.sin(3 * X) + X, 3.0 * np.cos(X) - X))
-    Ptrue: AnyNDArray = np.array([[0.5, -0.3, 1.5], [-0.4, 0.43, 0.0]])  # [L, P]
-
-    Y = tf.convert_to_tensor(G @ Ptrue)
-    G = tf.convert_to_tensor(np.hstack((0.5 * np.sin(3 * X) + X, 3.0 * np.cos(X) - X)))
-    Ptrue = tf.convert_to_tensor(np.array([[0.5, -0.3, 1.5], [-0.4, 0.43, 0.0]]))  # [L, P]
-    Y += tf.random.normal(Y.shape, dtype=tf.float64) * [0.2, 0.2, 0.2]
-    Xs = tf.convert_to_tensor(np.linspace(-6, 6, Ntest)[:, None])
+
+    X = cs(tf.random.normal((N,), dtype=tf.float64)[:, None] * 10 - 5, "[N, 1]")
+    G = cs(tf.concat([0.5 * tf.sin(3 * X) + X, 3.0 * tf.cos(X) - X], axis=1), "[N, L]")
+    Ptrue = cs(tf.constant([[0.5, -0.3, 1.5], [-0.4, 0.43, 0.0]], dtype=tf.float64), "[L, P]")
+    Y = cs(G @ Ptrue, "[N, P]")
+    Y += cs(tf.random.normal(Y.shape, dtype=tf.float64) * [0.2, 0.2, 0.2], "[N, P]")
```

Comment:
As much as I like the shape checking, maybe doing it on every line is a bit of overkill? 😄